src/hotspot/cpu/x86/gc/x/x_x86_64.ad

```diff
@@ -126,17 +126,15 @@ instruct xCompareAndSwapP(rRegI res, indirect mem, rRegP newval, rRegP tmp, rFla
 
   format %{ "lock\n\t"
             "cmpxchgq $newval, $mem\n\t"
-            "sete     $res\n\t"
-            "movzbl   $res, $res" %}
+            "sete_with_zextl  $res\n\t" %}
```

Comment:
Please, use `setcc` in format to match `ins_encode`.  `sete_with_zextl` isused only when it is supported.

Reply:
Hi @vnkozlov ,
setcc used in inst_encoding block is a macro assembly routine which emits either setcc + movzbl or setzucc for APX supported targets, I wanted to use one opto instruction to correctly depict semantics of both the cases.

Reply:
Yes, I see `setcc` definition and using it in `format` is fine since it will match to `inst_encoding`.
On other hand, there is no macro or assembler instruction `sete_with_zextl` and it will be confusing.

If you want you can add comment to format (and you should not use `\n` in last line):
```
   "setcc    $res\t#  emits sete + movzbl or setzucc for APX" %}
```

Reply:
DONE